android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/SignTransactionFragment.java

```diff
@@ -0,0 +1,364 @@
+                            updateSignDataAndDetails();
+                        });
+                break;
+            default: // Do nothing
+        }
+    }
+
+    private void updateTxPanelPerStep() {
+        if (mSignTxStep == SignTx.SIGN_RISK) {
+            mViewPager.setVisibility(View.GONE);
+            mTvTxCounter.setVisibility(View.GONE);
+            mBtnCounterNext.setVisibility(View.GONE);
+            mWarningLl.setVisibility(View.VISIBLE);
+            mBtSign.setBackgroundTintList(ColorStateList.valueOf(
+                    ContextCompat.getColor(requireContext(), R.color.brave_theme_error)));
+            mBtSign.setCompoundDrawablesWithIntrinsicBounds(0, 0, 0, 0);
+            mBtSign.setText(R.string.continue_text);
+        } else if (mSignTxStep == SignTx.SIGN_TX) {
+            mWarningLl.setVisibility(View.GONE);
+            mViewPager.setVisibility(View.VISIBLE);
+            mBtSign.setBackgroundTintList(ColorStateList.valueOf(
+                    ContextCompat.getColor(requireContext(), R.color.brave_action_color)));
+            mBtSign.setCompoundDrawablesRelativeWithIntrinsicBounds(R.drawable.ic_key, 0, 0, 0);
+            mBtSign.setText(R.string.brave_wallet_sign_message_positive_button_action);
+        }
+    }
+
+    private void processRequest(boolean isApproved) {
+        if (isApproved) {
+            // Continue clicked, update ui
+            if (mSignTxStep == SignTx.SIGN_RISK) {
+                mSignTxStep = SignTx.SIGN_TX;
+                updateTxPanelPerStep();
+                updateSignDataAndDetails();
+                return;
+            }
+        }
+        switch (mActivityType) {
+            case SIGN_TRANSACTION:
+                mWalletModel.getDappsModel().signTxRequest(isApproved, mSignTransactionRequest);
+                break;
+            case SIGN_ALL_TRANSACTIONS:
+                mWalletModel.getDappsModel().signAllTxRequest(
+                        isApproved, mSignAllTransactionsRequest);
+                break;
+            default: // Do nothing
+        }
+    }
+
+    private void updateAccount() {
+        BraveActivity activity = BraveActivity.getBraveActivity();
+        if (activity != null) {
+            activity.getWalletModel()
+                    .getKeyringModel()
+                    .getSelectedAccountOrAccountPerOrigin()
+                    .observe(getViewLifecycleOwner(), accountInfo -> {
+                        if (accountInfo == null) return;
+                        Utils.setBlockiesBitmapResource(
+                                mExecutor, mHandler, mAccountImage, accountInfo.address, true);
+                        String accountText = accountInfo.name + "\n" + accountInfo.address;
+                        mAccountName.setText(accountText);
+                    });
+        }
+    }
+
+    private void updateNetwork() {
+        getJsonRpcService().getNetwork(CoinType.ETH,
```

Comment:
do we hardcode it to `CoinType.ETH` only?

Reply:
fixed: https://github.com/brave/brave-core/pull/15389/commits/0c925e12fee38080c6ba602895d813a2d98e489d